services/packagist/packagist-dependency-version.service.js

```diff
@@ -0,0 +1,197 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { NotFound } from '../index.js'
+import {
+  allVersionsSchema,
+  BasePackagistService,
+  customServerDocumentationFragment,
+} from './packagist-base.js'
+
+const queryParamSchema = Joi.object({
+  dependencyVendor: Joi.string(),
+  dependencyRepo: Joi.string(),
+  server: optionalUrl,
+}).required()
+
+export default class PackagistDependencyVersion extends BasePackagistService {
+  static category = 'platform-support'
+
+  static route = {
+    base: 'packagist/dependency-v',
+    pattern: ':user/:repo/:version?',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Packagist Dependency Version Support',
+      pattern: ':user/:repo',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+      },
+      queryParams: {
+        dependencyVendor: 'twig',
+        dependencyRepo: 'twig',
+      },
+      staticPreview: this.render({
+        dependencyNameForLabel: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (specify version)',
+      pattern: ':user/:repo/:version',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        version: 'v2.8.0',
+      },
+      queryParams: {
+        dependencyVendor: 'twig',
+        dependencyRepo: 'twig',
+      },
+      staticPreview: this.render({
+        dependencyNameForLabel: 'twig/twig',
+        dependencyVersion: '1.12',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (custom server)',
+      pattern: ':user/:repo',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+      },
+      queryParams: {
+        dependencyVendor: 'twig',
+        dependencyRepo: 'twig',
+        server: 'https://packagist.org',
+      },
+      staticPreview: this.render({
+        dependencyNameForLabel: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+      documentation: customServerDocumentationFragment,
+    },
+  ]
+
+  static defaultBadgeData = {
```

Comment:
If we throw an error, the badge's label will show `dependency version` (instead of the dependency's name)